### ISSUE TITLE
PSYoungGen getDuration method not returning duration for a specific log

### ISSUE THREAD

**Issue by karianna** [microsoft/gctoolkit#224]
> Every event inherits from com.microsoft.gctoolkit.event.jvm.JVMEvent which is where getDuration() is defined. Therefore, every event has a getDuration() method.
> 
> I checked the generated javadoc for PSYoungGen and the methods inherited from JVMEvent are given. 

Yes, the method is inherited from JVMEvent. However that method does not return anything for PSYoungGen event. Similar is the case for many other events. By inheritence, it should have that data. But either the parsers are not able to capture those events, or there is something else wrong. 

For reference, i ran the same GC log file with GCToolkit and GCeasy. GCEasy was able to get PSYoungGen event duration, GCToolkit was not.

Let me know if you need the GC Log file.

_Originally posted by @HritwikSinghal in https://github.com/microsoft/gctoolkit/discussions/218#discussioncomment-3095648_

**Comment by karianna** [comment]
@HritwikSinghal - are you able to post the log here?

**Comment by HritwikSinghal** [comment]
@karianna yes, here it is. https://github.com/HritwikSinghal/gctoolkit/releases/download/log_file/hbase.gc.0.log

**Comment by HritwikSinghal** [comment]
Ok, so a little bit more info for debugging.  ![Uploading Screenshot from 2022-07-08 06-21-26.png…](Here) are stats from GCEasy for above log file. You can See that it can parse YoungGC and ConcurrentPreClean and give stats. 

However if i try to run the same GC log file with my GCToolkit sample script, it shows these values for PSYoungGen, YoungGC, ConcurrentPreClean and even for.

```
Total Time                  : 0.0 sec
Minimum duration:           :0.0 sec
Maximum duration:           :0.0 sec
```

So this means either something is wrong with my code, or the parsers are not able to parse these events.

The GCToolkit sample which i am running can be found [here](https://github.com/HritwikSinghal/gctoolkit/tree/test/sample). (Use the branch "test" ). 

The GCToolkit is able to parse InitialMark and ReMark stats properly (just uncomment respective lines in [this file](https://github.com/HritwikSinghal/gctoolkit/blob/test/sample/src/main/java/com/microsoft/gctoolkit/sample/aggregation/CMSTimeAggregator.java)).

**Comment by kcpeppe** [comment]
Thank you for all the extra information. We ha(ve)(d) test to ensure that the events contain the correct values. The first thing here is to check that those tests are being executed. I've been away from a computer for a few days but I'll be digging in this week.

**Comment by HritwikSinghal** [comment]
@karianna @kcpeppe any updates on this?

**Comment by karianna** [comment]
CC @kirk-microsoft for work inbox :-)
